src/views/credentials/index.tsx

```diff
@@ -0,0 +1,136 @@
+        return <p>Received</p>
+      case 'issuer':
+        return <p>Given</p>
+      default:
+        return <p>unknown</p>
+    }
+  }
+
+  private buttonText = (state: State): string => {
+    switch (state) {
+      case 'pending':
+        return 'View Request'
+      case 'done':
+        return 'View Credential'
+      default:
+        return 'default'
+    }
+  }
+
+  public listPage = (credentials: Credential[], search: string = '') => {
+    return (
+      <Page
+        title="Veritable - Credentials"
+        heading="Credential Management"
+        activePage="credentials"
+        headerLinks={[{ name: 'Credential Management', url: '/credentials' }]}
+        stylesheets={['query.css']}
+        hx-get="/credentials"
+        hx-trigger="every 10s"
+        hx-select="#search-results"
+        hx-target="#search-results"
+        hx-swap="outerHTML"
+        hx-include="#queries-search-input"
+      >
+        <div class="list-page-header">
+          <span>Credential Management</span>
+          <LinkButton
+            disabled={true}
+            text="New Credential"
+            href="/credentials/new"
+            icon={'url("/public/images/plus.svg")'}
+            style="filled"
+          />
+        </div>
+        <div class="list-page ">
+          <div class="list-nav">
+            <span>Credentials</span>
+            <input
+              id="queries-search-input"
+              class="search-window"
+              type="search"
+              name="search"
+              value={Html.escapeHtml(search)}
+              placeholder="Search"
+              hx-get="/credentials"
+              hx-trigger="input changed delay:500ms, search"
+              hx-target="#search-results"
+              hx-select="#search-results"
+              hx-swap="outerHTML"
+            ></input>
+          </div>
+          <table class="list-page">
+            <thead>
+              {['Company Name', 'Credential Type', 'Direction', 'Crendetial status', 'Actions'].map((name: string) => (
+                <th>
+                  <span>{name || 'unknown'}</span>
+                  <a class="list-filter-icon icon disabled" />
+                </th>
+              ))}
+            </thead>
+            <tbody id="search-results">
+              {credentials.length == 0 ? (
+                <tr>
+                  <td>No Credentials for that search. Try again or add a new credential</td>
+                </tr>
+              ) : (
+                credentials.map((cred) => {
+                  if (!cred || !cred.credentialAttributes) throw new Error('TODO update error')
+                  return (
+                    <tr>
+                      <td>{Html.escapeHtml(cred.credentialAttributes[0].value)}</td>
+                      <td>{Html.escapeHtml('Supplier credentials')}</td>
+                      <td>{this.roleToDirection(cred.role)}</td>
+                      <td>{statusToClass(cred.state as ConnectionStatus)}</td>
```

Comment:
This looks incorrect. Your passing a credential status as a connection status and re-using the same transform. Does that really work?

Reply:
I dropped all states into one, mmm.. it does but would need to confirm across the board. since 3 files have had their own (status/State).do you think I should revert?

Reply:
There's at least two places where `statusToClass` was duplicated so I don't mind you doing that. I'm more worried that this particular one seems wrong

Reply:
yes! we set a css class and text depending on state I think let's catch up on this